src/main/java/seedu/address/logic/commands/InternshipEditCommand.java

```diff
@@ -0,0 +1,274 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONTACT_NUMBER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROLE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.address.model.InternshipModel.PREDICATE_SHOW_ALL_INTERNSHIPS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.InternshipMessages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.InternshipModel;
+import seedu.address.model.internship.ApplicationStatus;
+import seedu.address.model.internship.CompanyName;
+import seedu.address.model.internship.ContactEmail;
+import seedu.address.model.internship.ContactName;
+import seedu.address.model.internship.ContactNumber;
+import seedu.address.model.internship.Description;
+import seedu.address.model.internship.Internship;
+import seedu.address.model.internship.Location;
+import seedu.address.model.internship.Role;
+
+/**
+ * Edits the details of an existing internship in the internship data.
+ */
+public class InternshipEditCommand extends InternshipCommand {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the internship identified "
+            + "by the index number used in the displayed internship data. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_COMPANY + "COMPANY] "
+            + "[" + PREFIX_CONTACT_NAME + "CONTACT_NAME] "
+            + "[" + PREFIX_CONTACT_EMAIL + "CONTACT_EMAIL] "
+            + "[" + PREFIX_CONTACT_NUMBER + "CONTACT_NUMBER] "
+            + "[" + PREFIX_LOCATION + "LOCATION] "
+            + "[" + PREFIX_STATUS + "STATUS] "
+            + "[" + PREFIX_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_ROLE + "ROLE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_EMAIL + "nussu@u.nus.edu"
+            + PREFIX_CONTACT_NUMBER + "9666 1666";
+
+    public static final String MESSAGE_EDIT_INTERNSHIP_SUCCESS = "Edited Internship: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_INTERNSHIP = "This person already exists in the internship data.";
+
+    private final Index index;
+    private final EditInternshipDescriptor editInternshipDescriptor;
+
+    /**
+     * @param index of the person in the filtered internship list to edit
```

Comment:
can go through all the javadocs to edit persons to internships, similar comments elsewhere